xos.h

```diff
@@ -144,13 +144,24 @@ namespace ast {
 class Expr {
  public:
   virtual ~Expr() = default;
+  bool operator==(
+      const Expr &other) const  // TODO(PiJoules): add rtti and implement the
+                                // comparison operator here.
+  {
+    const auto l = other;
```

Comment:
Remove this line. If you're running into an unused variable error, you can either add `[[maybe_unused]]` to the left of the argument or just remove the argument name.